### PR TITLE
Extract config package

### PR DESCRIPTION
**Pull request by magik6k**
Currently IPTB depends on go-ipfs directly which makes it hard to use it with anything that isn't go-ipfs (like go-ipfs-api) - https://github.com/ipfs/iptb/issues/48.

TODO:
* [x] Is this the right thing to do?
* [x] Push extracted repo to github - https://github.com/ipfs/go-ipfs-config
* [x] Fix/figure out how to manage `repo/config/version.go`
* [x] Update iptb to use `go-ipfs-config` - https://github.com/ipfs/iptb/pull/63

### CODE REVIEW

File: repo/config/version.go [hunk 1]
```diff
@@ -1,9 +0,0 @@
-package config
-
-// CurrentCommit is the current git commit, this is set as a ldflag in the Makefile
-var CurrentCommit string
-
-// CurrentVersionNumber is the current application's version literal
-const CurrentVersionNumber = "0.4.17-dev"
```
Comment (whyrusleeping) [inline repo/config/version.go]: Let's find a way to not have the binary version in a different package. I really don't think this stuff needs to be in the config package anyways

Comment (magik6k) [reply repo/config/version.go]: Yep, that definitely belongs here.

The problem is that we apparently embed that in config on init:
https://github.com/ipfs/go-ipfs/blob/80fcc4ecf177b3b6a8daa10559eb46bcf8a931db/repo/config/init.go#L27-L32

This was introduced in https://github.com/ipfs/go-ipfs/commit/feef5c3415ec249cf307f43a422545de24aee319#diff-22f880839db0df9b05dcad0718392fc9R28 which is in 0.4.16.

From what I can see we can safely drop that as in the only place it's used it's overridden with https://github.com/ipfs/go-ipfs/blob/master/core/corehttp/commands.go#L80

Comment (whyrusleeping) [reply repo/config/version.go]: That sounds good to me.

**Comment by whyrusleeping** [comment]
ci seems upset

**Comment by magik6k** [comment]
fixed

File: repo/version.go [hunk 2]
```diff
@@ -1,9 +1,9 @@
-package config
```
Comment (Stebalien) [inline repo/version.go]: This doesn't really feel like the right place to put this. Personally, I'd put it in the root along side `doc.go`.

**Comment by Stebalien** [comment]
One nit but, other than that (and a rebase once we cut a release), LGTM.

**Comment by whyrusleeping** [comment]
rebased and squashed this. Since i was at fault for the merge conflicts >.>

**Comment by kevina** [comment]
> Is this the right thing to do?

There didn't seam to be any discussion on this point.  It seams very odd to mean that we are putting the configuration for IPFS in an external package.  That is one thing that seams very IPFS specific.

**Comment by kevina** [comment]
> There are packages which depend on this (iptb to name one)

It just seams to be that there should be a better way, that's all.

**Comment by magik6k** [comment]
Needs a rebase now, but other than that it's good to go (I'll handle rebase tomorrow)

**Comment by Stebalien** [comment]
> It just seams to be that there should be a better way, that's all.

The config package could use some cleanup but allowing other programs to parse the config without pulling in all of go-ipfs seems generally useful.

**Comment by Stebalien** [comment]
@kevina although yeah, I kind of agree that this really does belong in this repo. However, we'll need to fix the gx situation for that to be a viable option.

Once we:

1. Make gx more pleasant to use.
2. Stop rewriting gx paths by default.

We can reevaluate this.

**Comment by whyrusleeping** [comment]
🔥 🔥 👌 🔥 👌  Go magik go!

**Pull request merged**